### PR TITLE
Add modalOnResponderTerminationRequest property

### PR DESCRIPTION
**Pull request by jdoyle65**
Was having issues where a custom Gesture Responder on one of my views was being triggered when scrolling up and down on the iOS Date Picker Modal. Added this property to override `onResponderTerminationRequest` callback on the Date Picker Modal.

By default the behaviour will be the same as before.

### CODE REVIEW

**Comment by feyy** [comment]
Awesome!

**Comment by feyy** [comment]
But I think we should set `modalOnResponderTerminationRequest` return default `false` to prevent parent component get responder, and add these properties to the picker msak.

**Comment by jdoyle65** [comment]
Actually when it returns `true` is when it will do the behaviour you're describing (a little bit counter-intuitive, for sure, but that's how it works in React Native).

**Comment by jdoyle65** [comment]
Actually, sorry I just realized I misread your comment. You're right, returning `false` would probably be a better default behaviour.

It's possible this would be a breaking change for anyone who has written workarounds to this in their own apps.

**Comment by feyy** [comment]
This feature will prevent the `TouchableHighlight`'s onPress event in my 7P, do you have some idea?

**Comment by jdoyle65** [comment]
According to the docs (https://facebook.github.io/react-native/docs/gesture-responder-system.html#responder-lifecycle) the default value this function returns is `false`.

Try passing in a function that returns true, and see if that works. RN will go to the deepest node first to see if it's trying to request being the responder. It will then bubble up to see if any of that node's parent/grandparents are also trying to become the responder. When `onResponderTerminationRequest` is set to return `false`, that Component will take over as the responder as it bubbles up, effectively stealing the responder from it's descendant.

So basically your `TouchableHighlight` requests to be the responder, but when the request bubbles up to the parent, the `View` Component that has the `onResponderTerminationRequest` property, it will steal the responder from `TouchableHighlight` if it's return `false`

**Comment by feyy** [comment]
I see. But this bug only grow in 7p, others work fine. And if we can only return `true` in `onResponderTerminationRequest`, this function can do so little！But I will return `true` fix this problem first!

**Comment by jdoyle65** [comment]
Yeah I find the gesture responder is a bit weird. Is it an OS level problem? Is your 7P on iOS 10 whereas the other devices are on iOS 9?

I found there's definitely slightly differing behaviour between gesture responders on the iOS 9 simulator and my physical device (Nexus 6P running Android 6.0.1)

**Comment by feyy** [comment]
I find the problem only grow in my 7p and 7p simulator, all with IOS 10. But the user report that his 6S also have this problem. Maybe it's OS version problem.

**Comment by jdoyle65** [comment]
@feyy I'm running into this issue with a tester who is using an iPhone 7 with iOS 10 as well. I have tested on an iPad Mini 2 running iOS 10.1.11 and it works, so it seems it may be device specific.

**Comment by hibearpanda** [comment]
@feyy I found this thread via this issue #50. I tested this component on an iphone 6s and iphone 6s plus and couldn't dismiss the modal via the Confirm, Cancel or the modal background easily without tapping it multiple times.

Just wanted to let you know regarding which models aren't working.

Using v1.3.1 right now.

**Comment by NeliHarbuzava** [comment]
Great PR. Very helpful. Thank you.

**Comment by ghost** [comment]
Great PR. Very helful. Thank you.
